app/controllers/reports_controller.rb

```diff
@@ -12,7 +12,12 @@ def create
   end
 
   def index
-    @reports = Report.all.limit(30)
+    if params[:latitude] && params[:longitude]
+      @reports = Report
+                 .near([params[:latitude], params[:longitude]])
+    else
+      @reports = Report.all.limit(30)
+    end
```

Comment:
Agree with @scmx.

We could have an endpoint `/reports/near`. But if both did use same pagination we could consider the params as search params and then maybe let them all be in the same action.